avalon/nuke/lib.py

```diff
@@ -0,0 +1,240 @@
+import os
+import contextlib
+import nuke
+import re
+import logging
+import toml
+
+log = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def maintained_selection():
+    nodes = nuke.allNodes()
+    previous_selection = [n.name()
+                          for n in nodes
+                          if n['selected'].value() is True]
+    try:
+        yield
+    finally:
+        # unselect all selection in case there is some
+        [n['selected'].setValue(False) for n in nodes]
```

Comment:
This would act "wrong" in the situation where during the context new nodes are created which weren't originally there. These would never get deselected since they are not in `nodes`. This should actually retrieve the current selection, and clear that?

Reply:
absolutely